source/Modules/ConfigMgrCBDsc.ResourceHelper/ConfigMgrCBDsc.ResourceHelper.psm1

```diff
@@ -514,11 +520,40 @@ function Get-ClientSettingsSoftwareCenter
     }
 }
 
+function Convert-CidrToIP
+{
+    [CmdLetBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [IPAddress]
+        $IPAddress,
+
+        [Parameter(Mandatory = $true)]
+        [ValidateRange(0,32)]
+        [Int32]
+        $Cidr
+    )
+
+    $CidrBits = ('1' * $Cidr).PadRight(32, '0')
+    $octets = $CidrBits -Split '(.{8})' -ne ''
+    $mask = ($octets | ForEach-Object -Process {[Convert]::ToInt32($_, 2) }) -Join '.'
+
+    $ip = [IPAddress](([IPAddress]"$IPAddress").Address -Band ([IPAddress]"$mask").Address)
```

Comment:
No need to typecast $IPAddress since it is delcared that in the parameters

Reply:
don't work correctly without the typecast